ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -193,20 +193,29 @@ private Long getEventFromPreviousDumpMetadata(Path previousDumpPath) throws Sema
   }
 
   private Path getPreviousDumpMetadataPath(Path dumpRoot) throws IOException {
+    FileStatus latestUpdatedStatus = null;
     FileSystem fs = dumpRoot.getFileSystem(conf);
     if (fs.exists(dumpRoot)) {
       FileStatus[] statuses = fs.listStatus(dumpRoot);
       if (statuses.length > 0)  {
```

Comment:
We don't need this check

Reply:
Is fs.listStatus() guaranteed to be not null even when the the path does not have any folder/files?

Reply:
It will return empty array

Reply:
Done